x-pack/plugins/enterprise_search/server/routes/enterprise_search/crawler/crawler.ts

```diff
@@ -86,9 +94,26 @@ export function registerCrawlerRoutes(routeDependencies: RouteDependencies) {
         });
       }
 
-      return enterpriseSearchRequestHandler.createRequest({
-        path: '/api/ent/v1/internal/indices',
-      })(context, request, response);
+      try {
+        await addConnector(client, connParams);
+        const res = await enterpriseSearchRequestHandler.createRequest({
+          path: '/api/ent/v1/internal/indices',
+        })(context, request, response);
+
+        if (res.status !== 200) {
+          throw new Error(res.payload.message);
+        }
+        return res;
+      } catch (error) {
+        // clean up connector index if it was created
+        const createdConnector = await fetchConnectorByIndexName(client, request.body.index_name);
+        if (createdConnector) {
+          await deleteConnectorById(client, createdConnector.id);
+          await deleteIndex(client, createdConnector.index_name);
+        }
+
+        throw error;
+      }
```

Comment:
Why do you create the connector, then the index, then delete the connect if index doesn't work?  Why not create index first then create connector only on success?  I might be missing something.

Reply:
Creating the connector can error out too! I think this is the correct order 🤔

Reply:
The ent-search endpoint also makes changes to the index created by calling the connector endpoint, so I need to call it in this order, otherwise it will be ent-search -> connector -> ent-search again.